sdk/src/azure/core/az_base64.c

```diff
@@ -0,0 +1,521 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <azure/core/az_base64.h>
+#include <azure/core/internal/az_precondition_internal.h>
+
+#include <azure/core/_az_cfg.h>
+
+// The maximum integer length of binary data that can be encoded into base 64 text and still fit
+// into an az_span which has an int32_t length, i.e. (INT32_MAX / 4) * 3;
+#define _az_MAX_SAFE_ENCODED_LENGTH 1610612733
+
+#define _az_ENCODING_PAD '='
+
+static const char _az_base64_encode_array[65]
+    = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+
+static const int8_t _az_base64_decode_array[256] = {
```

Comment:
I would try save some of that 256 bytes and try to implement it with ifs/switch-cases.

Reply:
That's an implementation detail, and not sure if it is worth it from a total code size perspective. I will investigate this in a follow-up:
https://github.com/Azure/azure-sdk-for-c/issues/1817